components/brave_rewards/resources/page/components/pageWallet.tsx

```diff
@@ -853,7 +855,7 @@ class PageWallet extends React.Component<Props, State> {
           this.state.modalVerify
             ? <ConnectWalletModal
                 rewardsBalance={balance.total}
-                providers={[{ type: walletType || '', name: walletProvider }]}
+                providers={externalWalletProviderList}
```

Comment:
If the rewards service gives us a list of wallet types, we can get the input we need here by doing something like:

```ts
externalWalletProviderList.map((type) => ({ type, name: utils.getWalletProviderName(type) })
```

If we do it this way, we'll have to update the `getWalletProviderName` function to accept either an ExternalWallet object or a string. Maybe something like:

```ts
export const getWalletProviderName = (wallet?: Rewards.ExternalWallet | string) => {
  if (!wallet) {
    return ''
  }
  switch (typeof wallet === 'string' ? wallet : wallet.type) {
    // etc.
  }
})
```

Reply:
resolved